### PR TITLE
Use System.Text.Json serializer by default.

### PR DESCRIPTION
**Pull request by ejball**
~~Using System.Text.Json is not trivial. Without objection I'm going to look into ways to make it easier to use System.Text.Json and harder to use Newtonsoft.Json.~~ Done.

### CODE REVIEW

**Comment by ejball** [comment]
@ddunkin Is `ServiceSerializer` really `ServiceJsonSerializer`? I'm trying to imagine how it could be used for anything but JSON.

**Comment by ddunkin** [comment]
The two existing implementations of `ServiceSerializer` use JSON, but a future implementation could use Protobuf.

**Review by ddunkin — APPROVED** [review]

**Comment by ejball** [comment]
> a future implementation could use Protobuf

I get the principle, but I don't get how `ServiceSerializer` could be used to achieve it. Every usage I can find is JSON-specific, and it has methods that couldn't be implemented by protobuf.

**Comment by ddunkin** [comment]
If one implemented a `ProtobufServiceSerializer`, `ProtobufHttpContentSerializer` (or modified `JsonHttpContentSerializer` to use a configurable content type), and added Protobuf metadata to the generated types, what would be JSON-specific?

**Comment by ejball** [comment]
I don't see how `ProtobufServiceSerializer` and `JsonServiceSerializer` would be interchangeable, so I don't see the point of a shared base, and knowing that `ProtobufServiceSerializer` can't do strings is a good indication that `ServiceSerializer` is not a good abstraction.

It just feels too speculative, and if we publish it, we're locked in. Your comment on `JsonHttpContentSerializer` is interesting; looking at its implementation, there really isn't anything JSON-specific about it. Perhaps if that were refactored it would be more obvious what the API for `ServiceSerializer` (`IServiceSerializer`?) should be.

**Comment by ejball** [comment]
Or maybe `HttpContentSerializer` is already the right abstraction. Why do we need another one?

**Comment by ddunkin** [comment]
Is the string representation of JSON useful for anything other than error or log messages? The only other use I see is in `ServiceDataUtility.Clone` and converting between representations in `ServiceObject`. Could `ServiceSerializer` `ToString`/`FromString` be replaced with `ToBytes`/`FromBytes`, which operate on a `Span<byte>` or similar? The JSON implementations would use UTF-8 bytes.

File: src/FacilityConformance/FacilityConformanceApp.cs [hunk 1]
```diff
@@ -84,7 +99,14 @@ public async Task<int> RunAsync(IReadOnlyList<string> args)
...
 				});
 
-			var tester = new ConformanceApiTester(m_tests, api, new HttpClient { BaseAddress = baseUri });
+			var tester = new ConformanceApiTester(
+				new ConformanceApiTesterSettings
+				{
+					Tests = tests,
+					Api = api,
+					HttpClient = new HttpClient { BaseAddress = baseUri },
+					ServiceSerializer = serializer,
```
Comment (ddunkin) [inline src/FacilityConformance/FacilityConformanceApp.cs:108]: `ServiceSerializer` should be set on `api`'s settings above.

**Review by ddunkin — COMMENTED** [review]
I understand standardizing on JSON for display strings, but I don't understand why it's better for the serializer abstraction to be JSON specific. It doesn't make the code better, and it makes it harder to support other serialization formats if we wanted that in the future, e.g. [MessagePack](https://github.com/FacilityApi/FacilityCSharp/pull/54#issuecomment-987641660).

**Comment by ejball** [comment]
If you are referring to the "rename `ServiceSerializer` to `JsonServiceSerializer`" commit, I reintroduced a simpler `ServiceSerializer` in a subsequent commit. I'll try to merge your MessagePack work into this PR.

**Review by ddunkin — APPROVED** [review]

**Pull request merged**